pandas/core/arrays/boolean.py

```diff
@@ -315,8 +315,13 @@ class BooleanArray(BaseMaskedArray):
     _TRUE_VALUES = {"True", "TRUE", "true", "1", "1.0"}
     _FALSE_VALUES = {"False", "FALSE", "false", "0", "0.0"}
 
-    def __init__(self, values: np.ndarray, mask: np.ndarray, copy: bool = False):
-        if not (isinstance(values, np.ndarray) and values.dtype == np.bool_):
+    def __init__(
+        self,
+        values: npt.NDArray[np.bool_],
+        mask: npt.NDArray[np.bool_],
+        copy: bool = False,
+    ):
+        if not (isinstance(values, np.ndarray) and values.dtype == np.dtype(np.bool_)):
```

Comment:
`values.dtype == np.dtype(np.bool_)` is slower than `values.dtype == np.bool_`, so if there is not a typing reason to do this, would prefer to keep it as is

Reply:
Added this because without it we end up with the mypy error 
```
error: Non-overlapping equality check (left operand type: "dtype[bool_]", 
right operand type: "Type[bool_]")  [comparison-overlap]
```

Wasn't sure of a great way around it, so used this pattern since have seen in a few other places. Could probably do something else like compare the dtype `kind` or `type`, but didn't want to deviate from existing code too much

Reply:
In that case, you can can maybe also define a `_dtype_bool = np.dtype(np.bool_)` top-level in this module, and compare against that (to avoid creating the dtype every time in the init here).

That's a pattern we also do elsewhere, for example: https://github.com/pandas-dev/pandas/blob/8f1dd29caf4ad954172c22f277ab336085566afc/pandas/core/internals/blocks.py#L123